modules/flann/include/opencv2/flann/dist.h

```diff
@@ -520,6 +520,7 @@ struct Hamming2
     unsigned int popcnt32(uint32_t n) const
     {
         n -= ((n >> 1) & 0x55555555);
+        n = ((n & 0xAAAAAAAA)>>1) | (n & 0x55555555);
```

Comment:
1. We should not confuse developers by misusing function names and implementations. There are a lot of popcnt optimizations, so this function body may be wrongly replaced in the future.
2. These two lines can be replaced to
```
n = (n | (n >> 1)) & 0x55555555;
```
3. Due possible HW optimizations of popcnt my suggestion is here: https://github.com/alalek/opencv/commit/issue_7989
4. Looks like OpenCV doesn't use this Hamming2 algorithm. Do you have a real failed example for this?
5. BTW, This code origin is [here](https://github.com/mariusmuja/flann/blame/master/src/cpp/flann/algorithms/dist.h), so it is better to send patch to upstream first. But looks like there is no Hamming2 distance at all in current upstream master.